### PR TITLE
Support running compatibility tests with the new data model.

### LINKED ISSUE #6473: Configure the model option (old versus new model) in OlderNessieServersExtension annotation
**Description**:
Would like to configure the model option in OlderNessieServersExtension annotation so that maven-failsafe-plugin is able to spin up the Nessie server with the correct model option instead of running the integration tests using verify for the new model where we have to run the OSS Nessie via maven apprunner plugin, which makes it harder to run/ debug tests in local.

For e.g:
mvn failsafe:integration-test  -pl :sample-model -Dit.test=ITSampleTest

**Requested Changes in public API**:
- Changes in REST API
- Changes in Java Client API
- Changes in Python Client API
- Changes in SQL extentions

**Expected Use Cases**:
To be able to spin up the Nessie server with new model using the OlderNessieServersExtension, so that failsafe-plugin can be used in Integration tests framework. This will allow us to debug/ run the specific test

### PR DESCRIPTION
**Pull request by dimas-b**
* Add server config property `nessie.test.storage.kind` with values `DATABASE_ADAPTER` (old) and `OBJECT` (new) to control the storage kind in the Nessie backend

* Add ITUnrelatedHistoryMerge to validate the new test framework capability and also validate that merges of unrelated histories are blocked in the model by default.

* "old server" code paths we validated manually with local artifacts. Proper tests for them will become possible after publishing a new Nessie version.

Fixes #6473

### CODE REVIEW

File: compatibility/common/src/main/java/org/projectnessie/tools/compatibility/internal/ServerKey.java [hunk 1]
```diff
@@ -52,19 +56,39 @@ public static ServerKey forContext(
...
+    StorageKind kind = StorageKind.DATABASE_ADAPTER;
+    if (storeKindStr != null) {
+      kind = StorageKind.valueOf(storeKindStr);
+    }
+
+    switch (kind) {
+      case DATABASE_ADAPTER:
+        return new ServerKey(version, databaseAdapterName, kind, config);
+
+      case OBJECT:
```
Comment (snazy) [inline compatibility/common/src/main/java/org/projectnessie/tools/compatibility/internal/ServerKey.java:70]: Hm - why not `PERSIST` as everywhere else?

File: compatibility/jersey/src/main/java/org/projectnessie/tools/compatibility/jersey/Backends.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+import org.projectnessie.versioned.storage.common.config.ImmutableAdjustable;
+import org.projectnessie.versioned.storage.common.config.StoreConfig;
+import org.projectnessie.versioned.storage.common.logic.Logics;
+import org.projectnessie.versioned.storage.common.persist.Backend;
+import org.projectnessie.versioned.storage.common.persist.BackendFactory;
+import org.projectnessie.versioned.storage.common.persist.Persist;
+import org.projectnessie.versioned.storage.common.persist.PersistFactory;
+import org.projectnessie.versioned.storage.common.persist.PersistLoader;
+
+public class Backends {
```
Comment (snazy) [inline compatibility/jersey/src/main/java/org/projectnessie/tools/compatibility/jersey/Backends.java:31]: Nit: `final` + ctor

File: version.txt [hunk 3]
```diff
@@ -1 +1 @@
-0.54.1-SNAPSHOT
\ No newline at end of file
+0.54.99
```
Comment (snazy) [inline version.txt:1]: ??

File: compatibility/common/src/main/java/org/projectnessie/tools/compatibility/internal/OlderNessieServersExtension.java [hunk 4]
```diff
@@ -56,7 +56,7 @@ private void populateFields(ExtensionContext context, Object instance) {
 
     BooleanSupplier initializeRepository = () -> true;
     ServerKey serverKey =
-        ServerKey.forContext(context, version, "In-Memory", Collections.emptyMap());
+        ServerKey.forContext(context, version, "In-Memory", "Inmemory", Collections.emptyMap());
```
Comment (snazy) [inline compatibility/common/src/main/java/org/projectnessie/tools/compatibility/internal/OlderNessieServersExtension.java:59]: I wouldn't mind if you change the name of the new in-memory to `In-Memory`, up to you.

**Review by snazy — COMMENTED** [review]
Mostly LGTM, a few nits and some strange change in verstion.txt

File: compatibility/common/src/main/java/org/projectnessie/tools/compatibility/internal/ServerKey.java (hunk 1 above)
Comment (dimas-b) [reply compatibility/common/src/main/java/org/projectnessie/tools/compatibility/internal/ServerKey.java:70]: sure, "object" relates to `ObjId` in `Persist`, but `PERSIST` LGTM too.

Comment (dimas-b) [reply compatibility/common/src/main/java/org/projectnessie/tools/compatibility/internal/ServerKey.java]: renamed

File: compatibility/jersey/src/main/java/org/projectnessie/tools/compatibility/jersey/Backends.java (hunk 2 above)
Comment (dimas-b) [reply compatibility/jersey/src/main/java/org/projectnessie/tools/compatibility/jersey/Backends.java]: fixed

**Review by snazy — APPROVED** [review]

**Pull request merged**